### PR TITLE
usage statistics reports config

### PR DESCRIPTION
**Pull request by gadomsky**

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/SettingsResources.scala [hunk 1]
```diff
@@ -86,6 +88,12 @@ object TopTabType extends Enumeration {
                              requiredPermission: Option[String],
                              addAccessTokenInQueryParam: Option[Boolean])
 
+private object FingerprintUtils {
+  lazy val random = Random.alphanumeric.take(10).mkString
+}
+
+@JsonCodec case class UsageReportsConfig(enabled: Boolean = true, fingerprint: String = FingerprintUtils.random)
```
Comment (mproch) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/SettingsResources.scala:95]: maybe enabled to defaultDesignerConfig?

**Review by mproch — APPROVED** [review]
Let's also add sth to changelog

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageReportsSettings.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+private object FingerprintUtils {
+  lazy val random = Random.alphanumeric.take(10).mkString
+}
+
+case class UsageReportsConfig(enabled: Boolean, fingerprint: String = FingerprintUtils.random)
+
+object UsageReportsSettings {
+  def apply(config: UsageReportsConfig): UsageReportsSettings = {
+    val version = Base64.getEncoder.encodeToString(BuildInfo.version.getBytes())
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageReportsSettings.scala:19]: I'd rather like to have here "sanitized" version but still user friendly. How would you like to present it/use in search on elastic? Let's say that you want all usages with version starting from 1.6.0.
By sanitization I mean e.g. all characters, numbers, .-_ - other things replaced by let's say "_"

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageReportsSettings.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import pl.touk.nussknacker.engine.version.BuildInfo
+
+import java.net.URLEncoder
+import java.util.Base64
+
+private object FingerprintUtils {
+  lazy val random = Random.alphanumeric.take(10).mkString
+}
+
+case class UsageReportsConfig(enabled: Boolean, fingerprint: String = FingerprintUtils.random)
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageReportsSettings.scala:15]: Let's add some prefix to easily recognize those randomly generated

**Review by arkadius — APPROVED** [review]

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageStatisticsReportsSettings.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import scala.util.Random
+import pl.touk.nussknacker.engine.version.BuildInfo
+
+import java.net.URLEncoder
+
+private object FingerprintUtils {
+  lazy val random = s"nu-${Random.alphanumeric.take(10).mkString}"
+}
+
+case class UsageStatisticsReportsConfig(enabled: Boolean, fingerprint: String = FingerprintUtils.random)
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageStatisticsReportsSettings.scala:14]: One more thing - IMO we should already add fingerprint let's say "development" for instances run from idea / runServer - it looks like env varaible would be nice for that

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageStatisticsReportsSettings.scala [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+import io.circe.generic.JsonCodec
+
+import scala.util.Random
+import pl.touk.nussknacker.engine.version.BuildInfo
+
+import java.net.URLEncoder
+
+private object FingerprintUtils {
+  lazy val random = s"nu-${Random.alphanumeric.take(10).mkString}"
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageStatisticsReportsSettings.scala:11]: maybe let this prefix be more meaningful. Nu is always true. Maybe Gen(erated)?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageStatisticsReportsSettings.scala (hunk 4 above)
Comment (gadomsky) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/UsageStatisticsReportsSettings.scala:15]: good point, I hope, I detected all places where we shoud override fingerprint:
 * runSever.sh
 * Run configuration in intellij
 * npm run backend:docker script
 
 do you have any idea if thats all ?

**Review by arkadius — APPROVED** [review]

**Pull request merged**